ipatests/test_xmlrpc/xmlrpc_test.py

```diff
@@ -207,18 +207,16 @@ class XMLRPC_test:
     """
     Base class for all XML-RPC plugin tests
     """
-
-    @classmethod
-    def setup_class(cls):
+    @pytest.fixture(autouse=True, scope="class")
+    def xmlrpc_setup(self, request):
         if not server_available:
-            raise unittest.SkipTest('%r: Server not available: %r' %
-                                (cls.__module__, api.env.xmlrpc_uri))
+            pytest.skip('%r: Server not available: %r' %
+                        (request.cls.__module__,
+                         api.env.xmlrpc_uri))
         if not api.Backend.rpcclient.isconnected():
             api.Backend.rpcclient.connect()
 
-    @classmethod
-    def teardown_class(cls):
-        request.destroy_context()
+        request.addfinalizer(lambda: ipa_request.destroy_context())
```

Comment:
You don't need the lambda here:
``request.addfinalizer(ipa_request.destroy_context)``

Reply:
@tiran, thank you.
I've replaced all the corresponding lambdas with the nested functions for uniformity reason.